substrafl_examples/get_started/run_diabetes_substrafl.py

```diff
@@ -0,0 +1,573 @@
+
+data_path = pathlib.Path.cwd() / "tmp" / "data_diabetes"
+data_path.mkdir(exist_ok=True)
+
+setup_diabetes(data_path=data_path)
+
+
+# %%
+# Registering data samples and dataset
+# ------------------------------------
+#
+# Every asset will be created in respect to predefined schemas (Spec) previously imported from
+# ``substra.sdk.schemas``. To register assets, :ref:`documentation/api_reference:Schemas`
+# are first instantiated and the specs are then registered, which generate the real assets.
+#
+# Permissions are defined when registering assets. In a nutshell:
+#
+# - Data cannot be seen once it's registered on the platform.
+# - Metadata are visible by all the users of a network.
+# - Permissions allow you to execute a function on a certain dataset.
+#
+# Next, we need to define the asset directory. You should have already downloaded the assets folder as stated above.
+#
+# A dataset represents the data in Substra. It contains some metadata and an *opener*, a script used to load the
+# data from files into memory. You can find more details about datasets
+# in the :ref:`API reference<documentation/references/sdk_schemas:DatasetSpec>`.
+
+from substra.sdk.schemas import DataSampleSpec
+from substra.sdk.schemas import DatasetSpec
+from substra.sdk.schemas import Permissions
+
+
+assets_directory = pathlib.Path.cwd() / "diabetes_substrafl_assets"
+assert assets_directory.is_dir(), """Did not find the asset directory,
+a directory called 'assets' is expected in the same location as this file"""
+
+permissions_dataset = Permissions(public=False, authorized_ids=DATA_PROVIDER_ORGS_ID)
+
+dataset = DatasetSpec(
+    name=f"Diabetes dataset",
+    type="csv",
+    data_opener=assets_directory / "dataset" / "diabetes_substrafl_opener.py",
+    description=data_path / "description.md",
+    permissions=permissions_dataset,
+    logs_permission=permissions_dataset,
+)
+
+# We register the dataset for each of the organizations
+dataset_keys = {
+    client_id: clients[client_id].add_dataset(dataset)
+    for client_id in DATA_PROVIDER_ORGS_ID
+}
+
+for client_id, key in dataset_keys.items():
+    print(f"Dataset key for {client_id}: {key}")
+
+
+# %%
+# The dataset object itself is an empty shell. Data samples are needed in order to add actual data.
+# A data sample contains subfolders containing a single data file like a CSV and the key identifying
+# the dataset it is linked to.
+
+datasample_keys = {
+    org_id: clients[org_id].add_data_sample(
+        DataSampleSpec(
+            data_manager_keys=[dataset_keys[org_id]],
+            test_only=False,
+            path=data_path / f"org_{i + 1}",
+        ),
+        local=True,
+    )
+    for i, org_id in enumerate(DATA_PROVIDER_ORGS_ID)
+}
+
+# %%
+# The data has now been added as an asset through the data samples.
+#
+# SubstraFL provides different type of Nodes to ingest these data, a Node being an object that will create an link
+# the different tasks with each other.
```

Comment:
I must admit, I had to re-read twice (admittedly, it's Friday and I got up waaay too early), and I contributed to the definition of those concepts 🙃 

It's just a suggestion but I think I would do the following: 
* I would start a new section at this point, after the data registration (which should be pretty straightforward for anyone familiar with Susbtra); _this_ is the important part of the tutorial, it should be easy to refer to it; this section will include the discussion of `Nodes` and of `ComputePlanBuilder`, maybe with two subsections.
* Move part of the "theoretical" explanation about compute graphs at the top of the section, to give some contexts why we are caring about nodes. Something like "a Federated computation can be represented as a graph of tasks, some of them executing on the data nodes (training tasks) and others aggregating local results (aggregation tasks). Substra does not store an explicit definition of this graph; instead, it gives the user full flexibility to define the computation graph they need, by linking a task to its parents (tasks that needs to be executed before this one, for example because it needs the outputs of the previous tasks). "
And then proceed with the discussion of the type of Nodes, and after of the `ComputePlanBuilder`. 

(I hope this is not too confusing, we can discuss it live if you want. Basically the idea is to first explain why it's important to set up the nodes, and how they relate with what the user really wants to do, that is building a Compute Graph. Then present the technical details. )